adp-aso-helm-library/templates/_flexible-servers-db.yaml

```diff
@@ -6,7 +6,7 @@ metadata:
   {{ $serverName := required (printf $requiredMsg "postgresServerName") .Values.postgresServerName }}
   {{ $dbName := required (printf $requiredMsg "postgres.db.name") .Values.postgres.db.name }}
   {{ $namespace := required (printf $requiredMsg "namespace") .Values.namespace }}
-  name: {{ printf "%s-%s-%s" $serverName $namespace $dbName }}
+  name: {{ (printf "%s-%s-%s" $serverName $namespace $dbName) | lower | replace "_" "-" }}
```

Comment:
@jeevankuduvaravindran : We dont need to do replace...dbname input is coming from user and we are expecting it as per allowed naming convention. I dont think you should transform it. if they pass '_' in naming then ideally it should fail